### ISSUE TITLE
[QUESTION] Returning multiple ids in batch insert operation

### ISSUE THREAD

**Issue by charlie87041** [jackc/pgx#1032]
I have this method which inserts multiple rows in a table and returns the pgx.Rows result
```
func (ptr *commons ) batchInsertAsset(wg *sync.WaitGroup) pgx.Rows{
	//defer ptr.Conn.Close()
	defer wg.Done()
	value, _ := ptr.pendingInsert.LoadAndDelete("generic_assets")
	batch := pgx.Batch{}
	for _, queable := range value.([]queableCode) {
		batch.Queue("INSERT INTO public.generic_assets( verified, subscription_date, code, internal_item_code,  acquisition_type_id, center_id, associated_surface_tag, associated_building_tag, validated,type_id,created_at, updated_at,  operational_state, temporal_id) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14) returning (id,temporal_id)",*queable.data...)
	}
	r := ptr.Conn.SendBatch(context.Background(), &batch)
	row,_ := r.Query()
	return row
}
```

Lately I access the row like this
```
//some method
var wg sync.WaitGroup
	wg.Add(1)
	ids := cache.batchInsertAsset(&wg)
	wg.Wait()

	debugger.chrono("imported assets at ")

	for ids.Next() != false{
		var a string
		var b int
		ids.Scan(&b, &a)


		fmt.Println(a)
		fmt.Println(b)
	}
```

So far so good. Thing is: it doesnt matter how many records I insert, the loop will always show me the first one. It doesnt matter if I run the batch with `Query()` or `QueryRow()`, response will always be the first record. What am missing here ???

Oh and great job with pgx, congrats

**Comment by jackc** [comment]
You need to run `Query` / `QueryRow` for each query you queued.

**Comment by henvic** [comment]
Hi @jackc,

I just ran into a similar issue with `Exec()` in a place where I don't really care about checking the result, and couldn't understand what is going on.

In the end, I solved it by calling the method n times the number of queries:

```go
for x := 0; x < batch.Len(); x++ {
	if _, err := br.Exec(); err != nil {
		return fmt.Errorf("cannot exec batch query: %w", err)
	}
}
```

Just before calling `br.Close()`.

I'm wondering if this is just a limitation of the current implementation or if this is supposed to be this way for some reason that is not clear to me.

Thanks!
